src/services/OperationalCalculationService.ts

```
@@ -0,0 +1,475 @@
+    }
+
+    if (flowRate < 0 && volumeDifference >= 0) {
+      // Emptying but already at or below setpoint
+      return null;
+    }
+
+    // Calculate time in minutes
+    const timeMinutes = Math.abs(volumeDifference / flowRate);
+
+    // Validate result
+    if (!isFinite(timeMinutes) || timeMinutes < 0) {
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)